docs/docs/reference/rendering-options/server-side-rendering.md

```diff
@@ -0,0 +1,133 @@
+
+Server-side Rendering (SSR) allows you to render a page at run-time with data that is fetched when a user visits the page.
+The server generates the full HTML during HTTP request and sends it to the user. The API is focused on data fetching outside of the Gatsby data-layer.
+
+## Creating server-rendered pages
+
+You can create server-rendered pages [the same way as regular pages](/docs/reference/routing/creating-routes/) (including using the File System Route API).
+
+The main difference is that page component must export an async function called `getServerData`:
+
+```js
+export async function getServerData(context) {
+  return {
+    props: {}, // Will be passed to the page component as "serverData" prop
+    headers: {}, // HTTP response headers for this page
+  }
+}
+```
+
+When a page component exports `getServerData` function, Gatsby treats all pages built with this component
+as server-rendered.
+
+The `context` parameter is an object with the following keys:
+
+- `headers`: The request headers
+- `method`: The request method, e.g. `GET`
+- `url`: The request URL
+- `query`: An object representing the query string
+- `params`: If you use [File System Route API](/docs/reference/routing/file-system-route-api/) the URL path gets passed in as `params`. For example, if your page is at `src/pages/{Product.name}.js` the `params` will be an object like `{ name: 'value' }`.
+
+`getServerData` can return an object with two keys:
+
+- `props` (optional): Object containing the data passed to `serverData` page prop. Should be a serializable object.
+- `headers` (optional): Object containing `headers` that are sent to the browser and caching proxies/CDNs, e.g. cache headers
+
+### Use `serverData` prop in React page component
+
+The `props` object you return from `getServerData` gets passed to the page component as `serverData` prop.
+
+```js
+import * as React from "react"
+
+const Page = ({ serverData }) => {
+  const { dogImage } = serverData
+
+  // Use dogImage in your page...
+}
+
+export async function getServerData() {
+  const res = await fetch(`https://a-dog-image.com/random`)
+  const data = await res.json()
+
+  return {
+    props: {
+      dogImage: data,
+    },
+  }
+}
+
+export default Page
+```
+
+## Interplay with build-time GraphQL queries
+
+Server-rendered pages also support regular Gatsby GraphQL page queries. The page query is executed at build-time
+and the data is passed to React component as `data` prop on each render (along with `serverData` prop).
+
+Keep in mind that `data` will be the same every time but `serverData` will change according to return value of your `getServerData` function.
+Runtime GraphQL queries are not supported yet.
+
+```js
+import * as React from "react"
+
+const Page = ({ data, serverData }) => {
+  const { site } = data
+  const { dogImage } = serverData
+  // Use dogImage and site info in your page...
```

Comment:
Similar suggestion here: I'd replace the comment with an actual short implementation.